storage/zynq7000-flash/flashdrv.c

```diff
@@ -398,126 +357,424 @@ ssize_t flashdrv_read(addr_t offs, void *buff, size_t len, time_t timeout)
 }
 
 
-ssize_t flashdrv_write(addr_t offs, const void *buff, size_t len)
+/* Block device interface */
+
+static int _flashdrv_sync(unsigned int id, size_t regStart)
 {
+	uint8_t *src;
+	addr_t dst;
 	ssize_t res;
-	time_t timeout;
-	size_t chunkSz = 0, freeSz = 0, saveSz = 0;
-	static const uint32_t timeoutFactor = 0x100;
-
-	uint32_t regID = 0, sectID = 0;
-	size_t sectSz, regStart;
+	unsigned int i, pgNb;
 
+	size_t pageSz, sectSz;
 	const flash_cfi_t *cfi = &fdrv_common.info.cfi;
 
-	if (len == 0)
-		return 0;
+	if (fdrv_common.regs[id].buffPos == 0) {
```

Comment:
Hmmm, why is this called buffPos? Why position zero means that there's nothing to do?

Reply:
`buffPos` define the current position in cache buffer.

It means that there is not any data in a buffer. For example, if `sync` function has been invoked two times, for the second time the buffer should not be saved to the flash memory.